simulation/utils/generate_map.py

```diff
@@ -55,82 +57,101 @@ def get_seed(self):
         """Get the seed of the map"""
         return self.seed
 
-    def X(self):
+    def get_X_list(self):
         """Get list of x coordinates"""
-        xs = [point.x for point in self.points_before_interpolation] if not len(self.map_points) else [point[0] for point in self.map_points]
+        if not len(self.map_points):
+            xs = [point.x for point in self.points_before_interpolation]
+        else:
+            xs = [point[0] for point in self.map_points]
```

Comment:
Dlaczego tu mamy point[0], a wyżej point.x?

Reply:
map_points i segment_points powstają dopiero przy interpolacji. Długość points_before_interpolation nie jest długa, więc posługuję się w nich moją klasą Point dla ułatwienia (chociaż nie wiem czy to nie troche przerost formy nad trescia juz xd) ale ze względu na optymalizację uznałam, że nie będę wypełniać tej dużo większej tablicy po interpolacji obiektami Point tylko zwykłymi tuplami. Stąc tutaj x, a tutaj 0.